### DISCUSSION TITLE
Temporary token

### DISCUSSION THREAD

**Discussion by pcostello-OD** [netbox-community/netbox#11822] [Ideas]
Any thoughts on introducing the creation of temporary tokens via api? We can create them currently, but they need to be manually deleted after use. Having a token that auto deletes itself after a set period of time would be nice.

**Comment by jeremystretch** [comment]
You can set an expiration time when creating a token:

![Screenshot 2023-02-24 at 12-19-36 Add a new token NetBox](https://user-images.githubusercontent.com/13487278/221245433-3b5c074a-c1b7-4f56-a17e-7ee9a7434cee.png)

The token won't be automatically deleted (to ensure it persists for audit purposes) but it will be considered invalid after the specified time.

**Reply by pcostello-OD** [reply]
Can you set the expiration when creating via the /users/tokens/provision/ endpoint?

**Reply by abhi1693** [reply]
Try posting to `/users/tokens/` using the data as

```json
{
  "user": 0,
  "expires": "2023-02-24T20:13:48.659Z",
  "last_used": "2023-02-24T20:13:48.659Z",
  "key": "stringstringstringstringstringstringstri",
  "write_enabled": true,
  "description": "string",
  "allowed_ips": []
}
```

Update the values as needed

**Reply by aahsing** [reply]
Does the value '0' userid?
